apps/middleman-workflows/src/lib/dal/activity.ts

```diff
@@ -0,0 +1,34 @@
+import { db } from "../db";
+import { Activity, activityTable } from "../db/schema";
+import { eq } from "drizzle-orm";
+
+export async function getActivity(activityId: number) {
+  return await db.query.activityTable.findFirst({
+    where: eq(activityTable.id, activityId),
+    with: { transactions: true },
+  });
+}
+
+export async function createActivity(activity: Activity) {
+  return await db
+    .insert(activityTable)
+    .values(activity)
+    .returning()
+    .then((res) => res[0]);
+}
```

Comment:
I don't think this will ever be used on in a workflow. I can't imaging right now a use case where we would automatically create a new activity. 
```suggestion
```